src/uct/tl/context.c

```diff
@@ -223,6 +227,38 @@ UCS_CLASS_DEFINE_NAMED_NEW_FUNC(uct_worker_create, uct_worker_t, uct_worker_t,
                                 ucs_async_context_t*, ucs_thread_mode_t)
 UCS_CLASS_DEFINE_NAMED_DELETE_FUNC(uct_worker_destroy, uct_worker_t, uct_worker_t)
 
+static ucs_status_t uct_config_read(uct_config_bundle_t **bundle,
+                                    ucs_config_field_t *config_table,
+                                    size_t config_size, const char *cfg_prefix,
+                                    const char *env_prefix)
+{
+    uct_config_bundle_t *config_bundle;
+    ucs_status_t status;
+
+    config_bundle = ucs_calloc(1, sizeof(*config_bundle) + config_size, "uct_config");
+    if (config_bundle == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
+
+    /* TODO use env_prefix */
+    status = ucs_config_parser_fill_opts(config_bundle->data, config_table,
+                                         env_prefix, cfg_prefix, 0);
+    if (status != UCS_OK) {
+        goto err_free_opts;
+    }
+
+    config_bundle->table = config_table;
+    config_bundle->table_prefix = cfg_prefix;
```

Comment:
i think we should use strdup here, too dangerous to assume user will not release the string memory